BrainPortal/app/models/cluster_task.rb

```diff
@@ -1661,9 +1661,14 @@ def unarchive_work_directory_from_userfile
       return false
     end
 
-    self.addlog("Attempting to restore TaskWorkdirArchive.")
+    if taskarch_userfile&.integrity_violated?
```

Comment:
invoke the method to compute the md5 and store it in the meta data of the task object, not the TarArchive, then compare at the moment of unarchiving

This context here already know about the status of the cache, so there is no need to invoke sync_to_cache at any time.

Reply:
ok